sdks/python/src/stepflow_py/flow_builder.py

```diff
@@ -385,14 +375,27 @@ def build(self) -> Flow:
                 "add_output_field() to specify the flow output."
             )
 
+        # Build FlowSchema if any schemas are set
+        schemas = None
+        if (
+            self.input_schema is not None
+            or self.output_schema is not None
+            or self.variables_schema is not None
+        ):
+            schemas = FlowSchema(
+                defs={},
```

Comment:
We may want to allow adding defs to the schema (eg., for things like ChatMessage which may be referenced in multiple steps). We may also want to support step schemas. Perhaps something like an optional `schema` argument when adding steps? And a dict of defs which can then be added to the defs list if new?